File: app/src/main/java/ru/roscha_akademii/medialib/storage/StorageTable.java
package ru.roscha_akademii.medialib.storage;

import ru.roscha_akademii.medialib.common.CreateTableQueryBuilder;

public class StorageTable {
    public static final int VERSION = 3;

    public static final String TABLE_NAME = "video_storage";

    public static final String REMOTE_URI = "remote_uri";
    public static final String STATUS = "status";
    public static final String LOCAL_URI = "local_uri";
    public static final String DOWNLOAD_ID = "download_id";
    public static final String PERCENT = "percent";

    public static String createTable() {
        return new CreateTableQueryBuilder(StorageTable.TABLE_NAME)
                .index(StorageTable.REMOTE_URI, CreateTableQueryBuilder.SqlType.STRING)
                .column(StorageTable.STATUS, CreateTableQueryBuilder.SqlType.INTEGER)
                .column(StorageTable.LOCAL_URI, CreateTableQueryBuilder.SqlType.STRING)
                .column(StorageTable.DOWNLOAD_ID, CreateTableQueryBuilder.SqlType.INTEGER)
                .column(StorageTable.PERCENT, CreateTableQueryBuilder.SqlType.INTEGER)
                .build();
    }
}

File: app/src/main/java/ru/roscha_akademii/medialib/storage/model/StorageTable.kt
package ru.roscha_akademii.medialib.storage.model

import ru.roscha_akademii.medialib.common.CreateTableQueryBuilder
import ru.roscha_akademii.medialib.common.CreateTableQueryBuilder.SqlType.STRING
import ru.roscha_akademii.medialib.common.CreateTableQueryBuilder.SqlType.INTEGER

object StorageTable {
    val VERSION = 6

    val TABLE_NAME = "video_storage"

    val REMOTE_URI = "remote_uri"
    val STATUS = "status"
    val LOCAL_URI = "local_uri"
    val DOWNLOAD_ID = "download_id"
    val PERCENT = "percent"

    fun createTable(): String {
        return CreateTableQueryBuilder(TABLE_NAME)
                .index(REMOTE_URI, STRING)
                .column(STATUS, INTEGER)
                .column(LOCAL_URI, STRING)
                .column(DOWNLOAD_ID, INTEGER)
                .column(PERCENT, INTEGER)
                .build()
    }
}
